lib/integer.gi

```diff
@@ -1215,13 +1258,11 @@ InstallGlobalFunction(RootInt,function ( arg )
     else Error("usage: `Root( <n> )' or `Root( <n>, <k> )'");
     fi;
 
-    # check the arguments and handle trivial cases
-    if  k <= 0                  then Error("<k> must be positive");
-    elif k = 1                  then return n;
-    elif n < 0 and k mod 2 = 0  then Error("<n> must be positive");
-    elif n < 0 and k mod 2 = 1  then return -RootInt( -n, k );
-    elif n = 0                  then return 0;
-    elif n <= k                 then return 1;
```

Comment:
This is the problem with the old code which caused it to accept some non-integer inputs: suppose `n <= k` holds (and none of the prior checks either handled the input, or indirectly caused an assertion, e.g. `k mod 2` will error out if `k` is a rational with even denominator). Then it returns 1. This is only sensible if also `1 <= n` holds, which is of course automatic if `n` is an integer (due to the preceding checks), but fails for rationals (such as e.g. `n=1/32`).

Since this "support" for rational inputs clearly was accidental (it's not documented; it's wrong for certain inputs; most rational inputs are not supported, e.g. `RootInt(4+1/2, 2);` always gave an error), I am strongly tempted to proceed with this PR as-is (targeting GAP 4.10, not 4.9), at least once a new `FactInt` version with the fix has been released.